message_ix_models/model/water/report.py

```diff
@@ -435,8 +447,148 @@ def prepare_ww(ww_input: pd.DataFrame, suban: bool) -> pd.DataFrame:
+    ]
+    non_cooling_water = [
+        v
+        for v in all_freshwater_tech
+        if not any(pattern in v for pattern in exclude_patterns)
+    ]
+
+    # Fresh water return flow emissions
+    fresh_return_emissions = report_iam.filter(
+        variable="emis|water_consumption|fresh_return|*|*"
+    ).variable
+
+    # Cooling investments
+    cooling_saline_inv = report_iam.filter(variable="inv cost|*saline").variable
+    cooling_air_inv = report_iam.filter(variable="inv cost|*air").variable
+    cooling_ot_fresh = report_iam.filter(variable="inv cost|*ot_fresh").variable
+    cooling_cl_fresh = report_iam.filter(variable="inv cost|*cl_fresh").variable
+
+    # Hydro electricity calculations
+    elec_hydro_var = report_iam.filter(variable="out|secondary|electr|hydro*").variable
+    report_iam = multiply_electricity_output_of_hydro(elec_hydro_var, report_iam)
+    water_hydro_var = report_iam.filter(
+        variable="Water Withdrawal|Electricity|Hydro|*"
+    ).variable
+
+    # Build cooling-specific mapping rows
+    cooling_rows = [
+        ["Water Withdrawal|Electricity|Hydro", water_hydro_var, "MCM/yr"],
+        [
+            "Water Withdrawal|Electricity|Cooling|Fresh Water",
+            cooling_fresh_water,
+            "MCM/yr",
+        ],
+        [
+            "Water Withdrawal|Electricity|Cooling|Once Through|Fresh Water",
+            cooling_ot_fresh_water,
+            "MCM/yr",
+        ],
+        [
+            "Water Withdrawal|Electricity|Cooling|Closed Loop|Fresh Water",
+            cooling_cl_fresh_water,
+            "MCM/yr",
+        ],
+        ["Water Withdrawal|Energy|Non-Cooling", non_cooling_water, "MCM/yr"],
+        ["Water Return|Electricity|Cooling", fresh_return_emissions, "MCM/yr"],
+        [
+            "Investment|Infrastructure|Water|Cooling",
+            cooling_ot_fresh + cooling_cl_fresh + cooling_saline_inv + cooling_air_inv,
+            "million US$2010/yr",
+        ],
+        [
+            "Investment|Infrastructure|Water|Cooling|Once through freshwater",
+            cooling_ot_fresh,
+            "million US$2010/yr",
+        ],
+        [
+            "Investment|Infrastructure|Water|Cooling|Closed loop freshwater",
+            cooling_cl_fresh,
+            "million US$2010/yr",
+        ],
+        [
+            "Investment|Infrastructure|Water|Cooling|Once through saline",
+            cooling_saline_inv,
+            "million US$2010/yr",
+        ],
+        [
+            "Investment|Infrastructure|Water|Cooling|Air cooled",
+            cooling_air_inv,
+            "million US$2010/yr",
+        ],
+    ]
+
+    # Store water_hydro_var for later filtering
+    report_iam.metadata = getattr(report_iam, "metadata", {})
+    report_iam.metadata["water_hydro_var"] = water_hydro_var
+    report_iam.metadata["cooling_inv_vars"] = (
+        cooling_ot_fresh + cooling_cl_fresh + cooling_saline_inv + cooling_air_inv
+    )
+
+    return report_iam, cooling_rows
```

Comment:
At the moment, there is no way to run only cooling and save that to a scenario, right?
I am not saying to do it now, maybe first we check with Oliver and maybe have an approach like to material (coupled with legacy, but running material new reporting)
Could you please write a comment with TODO, or so?